tests/test_splitter.py

```diff
@@ -0,0 +1,388 @@
+                                attributes=kwargs['attributes']))
+        def append_bbox_cvat(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                id=kwargs['ann_id'],
+                                attributes=kwargs['attributes'],
+                                group=kwargs['label_id'],
+                                z_order=kwargs['ann_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_labelme(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                attributes=kwargs['attributes'],
+                                id=kwargs['ann_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_mot(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_widerface(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, attributes=kwargs['attributes']))
+            annotations.append(Label(0, attributes=kwargs['attributes']))
+
+        functions = {
+            "coco": append_bbox_coco,
+            "voc" : append_bbox_voc,
+            "yolo" : append_bbox_yolo,
+            "cvat" : append_bbox_cvat,
+            "labelme" : append_bbox_labelme,
+            "mot": append_bbox_mot,
+            "widerface": append_bbox_widerface
+        }
+        params = []
+        for style in functions.keys():
+            for with_attr in [False, True]:
+                params.append((style, with_attr, 10, 5, 3, 2))
+                params.append((style, with_attr, 10, 7, 0, 3))
+
+        for style, with_attr, nimages, train, val, test in params:
+            with self.subTest(style=style, with_attr=with_attr, nimage=nimages,
+                              train=train, val=val, test=test):
+                source, _ = self._generate_detection_dataset(
+                            append_bbox = functions[style],
+                            with_attr=with_attr, nimages=nimages)
+                total = np.sum([train, val, test])
+
+                splits = [('train', train/total),
+                          ('val',   val/total),
+                          ('test',  test/total)]
+                actual = splitter.SplitforDetection(source, splits)
+                subsets = dict()
+                for sname in ['train', 'val', 'test']:
+                    subset = actual.get_subset(sname)
+                    subsets[sname] = subset
+                self.assertEqual(train, len(subsets['train']))
```

Comment:
Consider using `len(get_subset())` directly, as nothing else is needed for checking.

Reply:
Actually.. I need to check other values, but I can't, with the current statistics function.
Anyway, I'll change this, too. Thank you!